src/analysis/fourier/riemann_lebesgue_lemma.lean

```diff
@@ -9,175 +9,324 @@ import measure_theory.integral.integral_eq_improper
+
+/-- The Riemann-Lebesgue lemma for functions on `ℝ`, formulated via `real.fourier_integral`. -/
+theorem real.zero_at_infty_fourier_integral (f : ℝ → E) :
+  tendsto (𝓕 f) (cocompact ℝ) (𝓝 0) :=
+tendsto_integral_exp_inner_smul_cocompact f
+
+/-- Riemann-Lebesgue lemma for integrable functions, formulated via dual space.
+  **Do not use** -- it is only a stepping stone to `tendsto_integral_exp_smul_cocompact`. -/
+lemma tendsto_integral_exp_smul_cocompact_of_inner_product (μ : measure V) [μ.is_add_haar_measure] :
+  tendsto (λ w : V →L[ℝ] ℝ, ∫ v, e[-w v] • f v ∂μ) (cocompact (V →L[ℝ] ℝ)) (𝓝 0) :=
 begin
-  rw [real.cocompact_eq, tendsto_sup],
-  exact ⟨tendsto_integral_mul_exp_at_bot_of_continuous_compact_support hf1 hf2,
-    tendsto_integral_mul_exp_at_top_of_continuous_compact_support hf1 hf2⟩
+  obtain ⟨C, C_ne_zero, C_ne_top, hC⟩ := μ.is_add_haar_measure_eq_smul_is_add_haar_measure volume,
+  rw hC,
+  simp_rw integral_smul_measure,
+  rw ←(smul_zero _ : C.to_real • (0 : E) = 0),
+  apply tendsto.const_smul,
+  let A := (inner_product_space.to_dual ℝ V).symm,
+  have : (λ w : V →L[ℝ] ℝ, ∫ v, e[-w v] • f v) = (λ w : V, ∫ v, e[-⟪v, w⟫] • f v) ∘ A,
+  { ext1 w,
+    congr' 1 with v : 1,
+    rw [←inner_conj_symm, is_R_or_C.conj_to_real, inner_product_space.to_dual_symm_apply,
+      real.fourier_char_apply], },
+  rw this,
+  exact (tendsto_integral_exp_inner_smul_cocompact f).comp
+    A.to_homeomorph.to_cocompact_map.cocompact_tendsto',
 end
 
-open_locale fourier_transform
+end inner_product_space
 
-/-- Riemann-Lebesgue lemma for continuous compactly-supported functions: the Fourier transform
-tends to 0 at infinity. -/
-lemma real.fourier_integral_zero_at_infty_of_continuous_compact_support
-  (hc : continuous f) (hs : has_compact_support f) :
-  tendsto (𝓕 f) (cocompact ℝ) (𝓝 0) :=
+section no_inner_product
+
+variables
+  (f) [add_comm_group V] [topological_space V] [topological_add_group V] [t2_space V]
+  [measurable_space V] [borel_space V]
+  [module ℝ V] [has_continuous_smul ℝ V] [finite_dimensional ℝ V]
+  [complete_space E]
+
+/-- Riemann-Lebesgue lemma for integrable functions on a finite-dimensional real vector space,
+formulated via dual space. -/
+theorem tendsto_integral_exp_smul_cocompact (μ : measure V) [μ.is_add_haar_measure] :
+  tendsto (λ w : V →L[ℝ] ℝ, ∫ v, e[-w v] • f v ∂μ) (cocompact (V →L[ℝ] ℝ)) (𝓝 0) :=
 begin
-  refine ((zero_at_infty_integral_mul_exp_of_continuous_compact_support hc hs).comp
-    (tendsto_cocompact_mul_left₀
-    (mul_ne_zero (neg_ne_zero.mpr two_ne_zero) real.pi_pos.ne'))).congr (λ w, _),
-  rw [real.fourier_integral_eq_integral_exp_smul, function.comp_app],
-  congr' 1 with x:1,
-  ring_nf,
+  -- We have already proved the result for inner-product spaces, formulated in a way which doesn't
+  -- refer to the inner product. So we choose an arbitrary inner-product space isomorphic to V
+  -- and port the result over from there.
+  let V' := euclidean_space ℝ (fin (finrank ℝ V)),
```

Comment:
can you instead use `to_euclidean` from `analysis.inner_product_space.euclidean_dist`?

Reply:
I didn't want to presuppose a choice of normed-space structure on `V`, since a finite-dimensional real vector space has a canonical topology, but not a canonical choice of norm inducing that topology. Cf. comments a few lines further down about not using `nonempty_continuous_linear_equiv_of_finrank_eq`.

Reply:
I hadn't noticed that `to_euclidean` needed a normed space (it would perfectly make sense under your weaker assumptions). Given this, I agree that your approach is ok. Maybe add a `todo: refactor using to_euclidean once it does not need a normed structure any more`?